src/core/PasswordHealth.cpp

```diff
@@ -0,0 +1,153 @@
+/*
+ *  Copyright (C) 2019 KeePassXC Team <team@keepassxc.org>
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 or (at your option)
+ *  version 3 of the License.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <QApplication>
+#include <QString>
+
+#include "Database.h"
+#include "Entry.h"
+#include "Group.h"
+#include "PasswordHealth.h"
+#include "zxcvbn.h"
+
+PasswordHealth::PasswordHealth(double entropy)
+    : m_entropy(entropy)
+    , m_score(entropy)
+{
+    switch (quality()) {
+    case Quality::bad:
+    case Quality::poor:
+        m_reason = QApplication::tr("Very weak password");
+        m_details = QApplication::tr("Password entropy is %1 bit").arg(QString::number(m_entropy, 'f', 2));
+        break;
+
+    case Quality::weak:
+        m_reason = QApplication::tr("Weak password");
+        m_details = QApplication::tr("Password entropy is %1 bit").arg(QString::number(m_entropy, 'f', 2));
+        break;
+
+    case Quality::good:
+    case Quality::excellent:
+        // Reasons are essentially error messages; if there's nothing
+        // to complain, leave it empty.
+        break;
+    }
+}
+
+PasswordHealth::PasswordHealth(QString pwd)
+    : PasswordHealth(ZxcvbnMatch(pwd.toLatin1(), nullptr, nullptr))
+{
+}
+
+PasswordHealth::PasswordHealth(QSharedPointer<Database> db, QString pwd, Cache* cache)
+    : PasswordHealth(ZxcvbnMatch(pwd.toLatin1(), nullptr, nullptr))
+{
+    // Whenever the password is re-used, reduce score by
+    // this many points:
+    constexpr auto penalty = 15;
+
+    if (!db || !db->rootGroup())
```

Comment:
{ }

Reply:
Old habit, sorry :) fixed.

Can "make format" be configured to handle that automatically?